wurst/bitwise/Bitwise.wurst

```diff
@@ -0,0 +1,316 @@
+        result = this
+    return result
+
+public function int.byte1() returns int
+    return this mod 0x100
+
+public function int.byte2() returns int
+    return (this - (this div 0x10000) * 0x10000) div 0x100
+
+public function int.byte3() returns int
+    return (this - (this div 0x1000000) * 0x1000000) div 0x10000
+
+public function int.byte4(int isNegative) returns int
+    return isNegative * 128 + this div 0x1000000
+
+/** De-composes an integer into 4 unsigned bytes. **/
+public function int.bytes() returns byte4
+    var value = this
+    var isNegative = 0
+    if value < 0
+        value = -2147483648 + value
+        isNegative = 1
+
+    let b1 = value mod 0x100
+    let b2 = (value - (value div 0x10000) * 0x10000) div 0x100
+    let b3 = (value - (value div 0x1000000) * 0x1000000) div 0x10000
+    let b4 = isNegative * 128 + value div 0x1000000
+
+    return byte4(b1, b2, b3, b4)
+
+/**
+    Produces a signed integer from the bitstring.
+    Input string may contain whitespace for padding.
+**/
+public function fromBitString(string bitstring) returns int
+    var pow = 1
+    var result = 0
+    for i = (bitstring.length() - 1) downto 0
+        let char = bitstring.charAt(i)
+        if char != " "
+            if char == "1"
+                result += pow
+            pow *= 2
+
+    return result
+
+/**
+    Produces a bistring from this integer.
+**/ 
+public function toBitString(int num) returns string
+    var result = ""
+    var value = num
+
+    while value != 0
+        result = (value mod 2).toString() + result 
+        value = bwShiftr32(value, 1)
+
+    return result
+
+
+/** 4-bit AND operator **/
+function bwAnd4(int lhs, int rhs) returns int
+    return andData[lhs * 0x10 + rhs]
+
+/** 4-bit OR operator **/
+function bwOr4(int lhs, int rhs) returns int
+    return orData[lhs * 0x10 + rhs]
+
+/** 4-bit XOR operator **/
+function bwXor4(int lhs, int rhs) returns int
+    return xorData[lhs * 0x10 + rhs]
+
+/** 8-bit NOT operator **/
+function bNot8(int val) returns int
+    return 0xFF - val
+
+/** 8-bit AND operator **/
+function bwAnd8(int lhs, int rhs) returns int
+    return bwAnd4(lbyte[lhs], lbyte[rhs]) * 0x10 + bwAnd4(rbyte[lhs], rbyte[rhs])
+
+/** 8-bit OR operator **/
+function bwOr8(int lhs, int rhs) returns int
+    return bwOr4(lbyte[lhs], lbyte[rhs]) * 0x10 + bwOr4(rbyte[lhs], rbyte[rhs])
+
+/** 8-bit XOR operator **/
+function bwXor8(int lhs, int rhs) returns int
+    return bwXor4(lbyte[lhs], lbyte[rhs]) * 0x10 + bwXor4(rbyte[lhs], rbyte[rhs])
+
+/** 32-bit SHIFTL operator **/
+function bwShiftl32(int num, int shift) returns int
+    return num * powShift[shift]
+
+/** 32-bit SHIFTR operator **/
+function bwShiftr32(int num, int shift) returns int
+    var value = num
+    var isNegative = 0
+    if value < 0
+        value = -2147483648 + value
+        isNegative = 1
+    return isNegative * powShift[31-shift] + value div powShift[shift]
+
+// function bwRotl32(int val, int shift) returns int
```

Comment:
what's this commented stuff?

Reply:
bitwise rotation
it was in the library that I used as a reference, but it wasn't correctly working for negative numbers, so I commented it out - seeing as it wasn't really all that necessary anyway
I'm not sure if I can fix it